### PR TITLE
Refactor and add move constructor to DynamicBitset

### PR DESCRIPTION
**Pull request by roshandathathri**

### CODE REVIEW

File: libgalois/include/katana/DynamicBitset.h [hunk 1]
```diff
@@ -86,41 +86,49 @@ class KATANA_EXPORT DynamicBitset {
...
    */
-  size_t size() const { return num_bits; }
+  size_t size() const { return num_bits_; }
 
   /**
    * Gets the space taken by the bitset
    * @returns the space in bytes taken by this bitset
    */
-  // size_t alloc_size() const { return bitvec.size() * sizeof(uint64_t); }
+  // size_t alloc_size() const { return bitvec_.size() * sizeof(uint64_t); }
```
Comment (ddn0) [inline libgalois/include/katana/DynamicBitset.h]: delete?

File: libgalois/include/katana/DynamicBitset.h [hunk 2]
```diff
@@ -86,41 +86,49 @@ class KATANA_EXPORT DynamicBitset {
...
   /**
    * Shrinks the allocation for bitset to its current size.
    */
-  void shrink_to_fit() { bitvec.shrink_to_fit(); }
+  void shrink_to_fit() { bitvec_.shrink_to_fit(); }
+
+  /**
+   * Swaps the bitset with the given one
+   */
+  void swap(DynamicBitset& bitset) {
```
Comment (ddn0) [inline libgalois/include/katana/DynamicBitset.h]: Implementing `swap` directly is a slightly older pattern and more common pre C++11. If you did go this path, technically we should satisfy https://en.cppreference.com/w/cpp/types/is_swappable , which would require a free function `swap(DynamicBitset& a, DynamicBitset& b)`.

Easier to just implement move semantics (constructor and assignment) and not `swap` directly. `std::swap` will then just pick up on the moveability and do what you want.

Comment (roshandathathri) [reply libgalois/include/katana/DynamicBitset.h]: Ok, I'll add a move constructor instead of `swap`.

**Review by ddn0 — APPROVED** [review]

File: libgalois/include/katana/DynamicBitset.h [hunk 3]
```diff
@@ -43,41 +43,50 @@ namespace katana {
...
  **/
 class KATANA_EXPORT DynamicBitset {
-  katana::PODResizeableArray<katana::CopyableAtomic<uint64_t>> bitvec;
-  size_t num_bits{0};
+  katana::PODResizeableArray<katana::CopyableAtomic<uint64_t>> bitvec_{};
+  size_t num_bits_{0};
 
 public:
-  static constexpr uint32_t bits_uint64 = sizeof(uint64_t) * CHAR_BIT;
+  static constexpr uint32_t kNumBitsInUint64 = sizeof(uint64_t) * CHAR_BIT;
```
Comment (gurbinder533) [inline libgalois/include/katana/DynamicBitset.h:50]: `uint32_t` --> `size_t`?

**Review by gurbinder533 — APPROVED** [review]
LGTM.

File: libgalois/include/katana/DynamicBitset.h (hunk 3 above)
Comment (roshandathathri) [reply libgalois/include/katana/DynamicBitset.h:50]: As it's a small number, I was thinking about making it a `uint8_t` but didn't. It's a `constexpr`, so I don't think it matters except for readability. If you are strongly in favor of `size_t`, I can make it `size_t`.

Comment (gurbinder533) [reply libgalois/include/katana/DynamicBitset.h:50]: Yeah, I was just saying for consistency's sake. Don't worry abt it.

**Pull request merged**